action.yaml

```diff
@@ -52,6 +52,10 @@ inputs:
     description: Skip the installation of R package dependencies.
     required: false
     default: false
+  max-iter:
+    description: Maximum number of iterations to resolve dependencies.
+    required: false
+    default: 50
```

Comment:
not exactly a retry mechanism
if there is an error in dependency solver then we take erroneous packages, check against provided references (i.e. `lookup-refs` parameter) and add start the next iteration
This is to account for feature branch match for indirect dependencies, e.g. `teal` -imports-> `teal.widgets` -imports-> `rtables`. 

For us, I would expect that just a few iterations will be needed. Definitely less than 10.

Note that each of the iterations should be relatively fast. The high value of 50 is to secure against an infinite loop in case we don't correctly identify break condition (which is what this PR also does)

We have one great example here: https://github.com/insightsengineering/citril/actions/runs/10437589102/job/28903940741
The whole workflow lasted for about 6h (21600 secs). Take all the other steps away, there are ~5k occurrences of "Failed to resolve dependencies" phrase printed for each of the iteration. That would mean that each of the iteration lasted ~around~ less than 4.3 secs. Hence, there is no big difference between 10 or 20 or 50

Reply:
ok I see, let's change to 10 then. At the end of the day this is a default value of a. parameter so it can be overwritten with user input.
```suggestion
    default: 10
```